src/main/java/org/jetbrains/research/groups/ml_methods/refactoring/detection/DefaultBranchesDetectionTool.java

```diff
@@ -0,0 +1,55 @@
+package org.jetbrains.research.groups.ml_methods.refactoring.detection;
+
+import org.apache.log4j.Logger;
+import org.eclipse.egit.github.core.RepositoryId;
+import org.eclipse.egit.github.core.service.RepositoryService;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.research.groups.ml_methods.Logging;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.jetbrains.research.groups.ml_methods.refactoring.detection.RefactoringDetectionApplication.printExceptionInformation;
+
+abstract class DefaultBranchesDetectionTool implements RefactoringDetectionTool {
+    @NotNull
+    private static final Logger LOGGER = Logging.getLogger(DefaultBranchesDetectionTool.class);
+
+    @NotNull
+    private static String getDefaultBranch(@NotNull URL repositoryUrl) throws IOException {
+        RepositoryId repositoryId = RepositoryId.createFromUrl(ParsingUtils.getHttpLink(repositoryUrl));
+        return new RepositoryService().getRepository(repositoryId.getOwner(), repositoryId.getName()).getMasterBranch();
+    }
+
+    @NotNull
+    @Override
+    public List<RepositoryDetectedRefactorings> detect(@NotNull List<URL> repositoryUrls) {
+        String passedProjects = repositoryUrls.stream()
+                .map(ParsingUtils::getProjectName)
+                .collect(Collectors.joining(", "));
+        LOGGER.info("Started detection for projects: " + passedProjects);
+        List<RepositoryDetectedRefactorings> detected = new ArrayList<>();
+        int passedRepositories = 0;
+        for (URL repositoryUrl : repositoryUrls) {
+            System.out.println("Processed repositories: " + passedRepositories++ + " / " + repositoryUrls.size());
```

Comment:
It prints: "Processed repositories: 0 / 2". Not consistent with commits counting.